File: src/api/table-editor.ts
import { invokeCommand } from '@/api/base'
import type { TableDefinition, TableAlteration, DdlResult, TableDetail } from '@/types/table-editor'

export function generateCreateTableSql(definition: TableDefinition, driver: string): Promise<DdlResult> {
  return invokeCommand('generate_create_table_sql', { definition, driver })
}

export function generateAlterTableSql(alteration: TableAlteration, driver: string): Promise<DdlResult> {
  return invokeCommand('generate_alter_table_sql', { alteration, driver })
}

export function executeDdl(connectionId: string, sql: string): Promise<boolean> {
  return invokeCommand('execute_ddl', { connectionId, sql })
}

export function getTableDetail(connectionId: string, database: string, table: string): Promise<TableDetail> {
  return invokeCommand('get_table_detail', { connectionId, database, table })
}

export function getTableDdl(connectionId: string, database: string, table: string): Promise<string> {
  return invokeCommand('get_table_ddl', { connectionId, database, table })
}

File: src-tauri/src/commands/table_editor.rs
use tauri::{command, Manager};

use crate::commands::db::DbEngineState;
use crate::models::table_editor::{DdlResult, TableAlteration, TableDefinition, TableDetail};
use crate::services::table_editor;
use crate::utils::error::AppError;

#[command]
pub fn generate_create_table_sql(
    definition: TableDefinition,
    driver: String,
) -> Result<DdlResult, AppError> {
    table_editor::generate_create_table(&definition, &driver)
        .map_err(|e| AppError::Other(e))
}

#[command]
pub fn generate_alter_table_sql(
    alteration: TableAlteration,
    driver: String,
) -> Result<DdlResult, AppError> {
    table_editor::generate_alter_table(&alteration, &driver)
        .map_err(|e| AppError::Other(e))
}

#[command]
pub async fn execute_ddl(
    app: tauri::AppHandle,
    connection_id: String,
    sql: String,
) -> Result<bool, AppError> {
    let engine = app.state::<DbEngineState>().inner().clone();
    engine.execute_query(connection_id, None, sql, None)
        .await?;
    Ok(true)
}

#[command]
pub async fn get_table_detail(
    app: tauri::AppHandle,
    connection_id: String,
    database: String,
    table: String,
) -> Result<TableDetail, AppError> {
    let engine = app.state::<DbEngineState>().inner().clone();
    let pool = engine.get_pool(connection_id).await?;
    table_editor::get_table_detail(&pool, &database, &table)
        .await
}

#[command]
pub async fn get_table_ddl(
    app: tauri::AppHandle,
    connection_id: String,
    database: String,
    table: String,
) -> Result<String, AppError> {
    let engine = app.state::<DbEngineState>().inner().clone();
    let pool = engine.get_pool(connection_id).await?;
    table_editor::get_table_ddl(&pool, &database, &table)
        .await
}
